vllm_spyre/perf_metrics.py

```diff
@@ -0,0 +1,59 @@
+""" Spyre performance metric logging """
+import os
+import time
+
+import vllm_spyre.envs as envs
+
+
+def create_perf_metric_logger(rank: int):
+    """ Create a performance metric logging object. """
+    if envs.VLLM_SPYRE_PERF_METRIC_LOGGING_ENABLE == 1:
+        return SpyrePerfMetricFileLogger(rank)
+    return SpyrePerfMetricLoggerBase(rank)
+
+
+class SpyrePerfMetricLoggerBase:
+    """ A no-op base class for use when logging is disabled """
+
+    def __init__(self, rank: int):
+        self.rank = rank
+
+    def __del__(self):
+        pass
+
+    def log(self, description: str, value, **kwargs):
+        """ Log value with description. kwargs is used as a dictionary of
+            additional labels to further describe the logged value. """
+        pass
+
+
+class SpyrePerfMetricFileLogger(SpyrePerfMetricLoggerBase):
+    """ A per-rank file logging object """
+
+    def __init__(self, rank: int):
+        super().__init__(rank)
+        self.time_fmt = "%m-%d %H:%M:%S"
+        self.log_path = os.path.join(envs.VLLM_SPYRE_PERF_METRIC_LOGGING_DIR,
+                                     f"perf_log_rank_{str(rank)}.txt")
+        # Note, this file is closed during object deletion.
+        self.file = open(self.log_path, "w", buffering=1)  # noqa: SIM115
```

Comment:
Since this isn't something that we're writing a whole lot of info to (ie we're not putting one line in here per processed request) I think it's probably okay to open as needed with `"a"` mode to append each time, instead of keeping it open